gadget/update.go

```diff
@@ -76,6 +77,13 @@ type ContentObserver interface {
 		targetRootDir, sourcePath, relativeTargetPath string) (bool, error)
 }
 
+type ContentInstallObserver interface {
+	ContentObserver
+	// SetEncryptionKey stores the encrypted data partition key to be sealed
+	// at the end of the installation process.
+	SetEncryptionKey(secboot.EncryptionKey)
```

Comment:
nitpick: given the struct is called an Observer, I think a name like ChosenEncryptionKey(key) would be better for this?

Reply:
Yes, I agree. I was a bit uncertain about the generic setter name too.